index.html

```diff
@@ -3,134 +3,453 @@
+      .call(xAxis2);
+
+    svg2.append("g")
+      .attr("class","y axis")
+      .attr("transform","translate("+ padding + ",0)")
+      .call(yAxis2)
+      ;
+
+    svg2.append("text")
+      .attr("text-anchor", "middle")  // this makes it easy to centre the text as the transform is applied to the anchor
+      .attr("class","axislabel")
+      .attr("transform", "translate("+ (-axispadding) +","+(height/2)+")rotate(-90)")  // text is drawn off the screen top left, move down and out and rotate
+      .text("Arrests per 100,000 people");
+
+    svg2.append("text")
+      .attr("text-anchor", "middle")  // this makes it easy to centre the text as the transform is applied to the anchor
+      .attr("class","axislabel")
+      .attr("transform", "translate("+ (width/2) +","+(height+(xaxispadding))+")")  // centre below axis
+      .text("Median Income");
+
+    // EVENT HANDLERS
     d3.select('#year-range').on('change', function () {
-      options.yearRange = d3.event.target.value;
+      options.state = d3.event.target.value;
+      options.region = 'all';
+      console.log(xAxis.ticks());
+      update();
+    });
+
+    d3.select('#region-select').on('change', function () {
+      options.region = d3.event.target.value;
+      options.state = 'all';
       update();
     });
   }
 
+  function customAxis(g) {
+  g.selectAll("text")
+      .attr("x", 4)
+      .attr("dy", -4);
+}
+
   function update() {
     // TRANSFORM
     txData = data.slice();
-
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
+    if (options.state !== 'all') {
+      var state = options.state;
       txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
+        return d.Abbr === state;
       });
     }
-
+    if (options.region !== 'all') {
+      var region = options.region;
+      txData = txData.filter(function (d) {
+        return d.Region === region;
+      });
+    }
+    
     // SCATTERPLOT
+    var points = svg.selectAll('.point')
+      .data(txData);
+
+    points.enter().append('circle')
+      .attr('class', 'point');
+
+    points
+      .attr('cx', function (d) { return x(d.PovertyRate/100); })
+      .attr('cy', function (d) { return y(d.rate); })
+      .attr('r', function (d) { return rscale(d.Pop); });
+    points.exit().remove();
+
+    //2nd Scatterplot
+     var points2 = svg2.selectAll('.point2')
+      .data(txData);
 
-    // HISTOGRAMS
+    points2.enter().append('circle')
+      .attr('class', 'point2');
+
+    points2
+      .attr('cx', function (d) { return x2(d.MedianIncome); })
+      .attr('cy', function (d) { return y(d.rate); })
+      .attr('r', function (d) { return rscale(d.Pop); });
+    points2.exit().remove();
+
+
+
+      // HISTOGRAMS
+     var bins = d3.layout.histogram()
+      .bins(x.ticks().length-1)
+      (txData);
+
+    var bar = svg.selectAll('.bar')
+      .data(bins)
+      .enter().append('g')
```

Comment:
Note that because we immediately call `.enter()` after joining the data, the `bars` variable contains just the enter selection. This means that once we have added all of the bars the first time `update()` is called, we'll never change them again — there's always the same number of bars, so nothing to enter, thus all of this code is called on an empty selection the second and subsequent times it runs. We would instead call `bars.enter()...` on a new line so that we can continue to access the full `bars` selection.

Compare and contrast to how `points` is done above.

A group-specific note, although we could just get rid of the groups and create rects directly: we would need to append the rects in the enter selection, when we first create the groups. Otherwise, we'll just keep adding new rects every time through. Then in order to modify the rect inside each of the bar groups, we would use something like: `bars.selectAll('rect').attr(...)`